numpy/core/src/multiarray/datetime.c

```diff
@@ -1160,7 +1160,7 @@ get_datetime_conversion_factor(PyArray_DatetimeMetaData *src_meta,
     }
 
     /* If something overflowed, make both num and denom 0 */
-    if (denom == 0 || num == 0) {
+    if (num == 0) {
```

Comment:
The comment suggests that this is about overflows.  I am not sure that makes any sense, since overflows should not result in a 0 value reliably?  But I don't think removing `denom == 0` makes this any better.

At this point, I don't think an overflow can occur (i.e. the check is bogus).  The code was probably reorganized at some point.  Below we have `num *= ...` and `denom *= ...`. Both of those can probably overflow, and `npy_mul_with_overflow_int` would probably be good (I am not sure we have the `mul_with_overflow` for `int`.

Reply:
Without looking into `get_datetime_units_factor()` internals, it is documented to return 0 in case of overflow:
https://github.com/numpy/numpy/blob/26ce52951bf176ffb30c1044414036b728e0849d/numpy/core/src/multiarray/datetime.c#L1030-L1038
I think that's the case being addressed here.

I seem to recall `int` was 64-bit on Alpha processors (ILP64 data model).

Reply:
Ah OK, so this is only to catch overflow from `get_datetime_units_factor()`.  That makes sense then, thanks.